pkg/cloudprovider/providers/azure/azure_loadbalancer.go

```diff
@@ -496,6 +506,24 @@ func getIdleTimeout(s *v1.Service) (*int32, error) {
 	return &to32, nil
 }
 
+func judgeDeleteFrontendIPConfiguration(config network.FrontendIPConfiguration, service *v1.Service, lbFrontendIPConfigName string) bool {
+	if serviceOwnsFrontendIP(config, service) && !strings.EqualFold(to.String(config.Name), lbFrontendIPConfigName) {
+		return true
+	}
+	if !strings.EqualFold(to.String(config.Name), lbFrontendIPConfigName) {
+		return false
+	}
+	loadBalancerIP := service.Spec.LoadBalancerIP
```

Comment:
It should also be checked with config.PublicIPAddress. If loadBalancerIP == config.PublicIPAddress, then the frontendIPConfiguration should be reserved.